### PR TITLE
Remove old Windows runner and replace Windows Gradle Check Runner AMI ID with M58xlarge

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Remove old Windows runner and replace Windows Gradle Check Runner AMI ID with M58xlarge

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/3816

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/compute/agent-nodes.ts [hunk 1]
```diff
@@ -184,19 +182,6 @@ export class AgentNodes {
       initScript: 'echo',
       remoteFs: '/var/jenkins',
     };
-    this.WINDOWS2019_X64 = {
```
Comment (prudhvigodithi) [inline lib/compute/agent-nodes.ts:187]: We no longer use this @peterzhuamazon ?

**Review by prudhvigodithi — APPROVED** [review]

File: lib/compute/agent-nodes.ts [hunk 2]
```diff
@@ -234,13 +219,13 @@ export class AgentNodes {
     this.WINDOWS2019_X64_GRADLE_CHECK = {
       agentType: 'windows',
       customDeviceMapping: '/dev/sda1=:300:true:::encrypted',
-      workerLabelString: 'Jenkins-Agent-Windows2019-X64-C524xlarge-Single-Host',
-      instanceType: 'C524xlarge',
+      workerLabelString: 'Jenkins-Agent-Windows2019-X64-M58xlarge-Single-Host',
+      instanceType: 'M58xlarge',
```
Comment (prudhvigodithi) [inline lib/compute/agent-nodes.ts:223]: Whey are we changing the instance type, is C5 type not stable ?

File: lib/compute/agent-nodes.ts (hunk 1 above)
Comment (peterzhuamazon) [reply lib/compute/agent-nodes.ts:187]: Nope, we have since changed to windows docker in 2.10.0.

File: lib/compute/agent-nodes.ts (hunk 2 above)
Comment (peterzhuamazon) [reply lib/compute/agent-nodes.ts:223]: Yes, c524large is 1:2 cpu:mem while m58xlarge is 1:4 cpu:mem, which linux has already replace with.
Runs better, more stable, and less resource.

* #321

**Pull request merged**